relay-server/src/actors/events.rs

```diff
@@ -1461,7 +1472,15 @@ impl Handler<HandleEnvelope> for EventManager {
                     None => Err(ProcessingError::RateLimited(checked.rate_limits)),
                 }
             }))
+            .and_then(|envelope| {
+                sample_transaction(envelope, sampling_project, false).then(|result| match result {
```

Comment:
_nit_: This `.then` can be moved to the outer future chain.

Reply:
not sure what are you proposing, it is already at the top level of a long future chain.